### ISSUE TITLE
Add an ability to exclude sources from analysis

### ISSUE THREAD

**Issue by baev** [xvik/gradle-quality-plugin#2]

**Comment by xvik** [comment]
I assume that you want something like this:
```groovy
quality {
    exclude  = [
               '**/com/mypackage/badcode/**'
    ]
}
```
Is it correct? If not please show your example.

**Comment by baev** [comment]
yep, that is correct in general. My sourceSet contains two folders `build/generated-sources` and `src/main/java` (and I can split them to different source sets for some reason). So it would be nice to exclude generated sources from analysis (generated sources is in the same package and may contains a lot of classes)

**Comment by baev** [comment]
@xvik is there any chance to get this implemented soon?

**Comment by xvik** [comment]
It's already implemented but needs some polishing before publication. 
The new version should be released in a couple of days.

**Comment by baev** [comment]
@xvik glad to hear, thanks

**Comment by xvik** [comment]
Released [2.2.0](https://github.com/xvik/gradle-quality-plugin#direct-source-exclusions)

I suppose your configuration is something like this:

```groovy
sourceSets.main {
    java {
        srcDir 'build/generated-sources'
    }
}
```

So to exclude all generated classes you can use:

```groovy
quality {
    excludeSources = fileTree('build/genearted-sources')
}
```

See full description in [the new docs chapter](https://github.com/xvik/gradle-quality-plugin#exclude-files-from-check)

**Comment by baev** [comment]
thank you so much
